lib/rubygems/commands/build_command.rb

```diff
@@ -26,6 +26,8 @@ def initialize
     add_option "-C PATH", "Run as if gem build was started in <PATH> instead of the current working directory." do |value, options|
       options[:build_path] = value
     end
+    deprecate_option "-C",
+                     extra_msg: "-C is a global flag now. Use `gem -C PATH build GEMSPEC_FILE [options]` instead"
```

Comment:
If you guys want, I can add a version here.

Reply:
Let's say 4.0 I guess.